MacOS build failure in video.cpp

```
 [157/269] Building CXX object CMakeFiles/DDNet.dir/src/engine/client/video.cpp.o
FAILED: CMakeFiles/DDNet.dir/src/engine/client/video.cpp.o 
/Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCONF_DEBUG -DCONF_GLEW_HAS_CONTEXT_INIT -DCONF_INFORM_UPDATE -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"16.2.2\" -DGLEW_STATIC -I/Users/runner/work/ddnet/ddnet/debug/src -I/Users/runner/work/ddnet/ddnet/src -isystem /usr/local/opt/freetype/include/freetype2 -isystem /Users/runner/work/ddnet/ddnet/src/engine/external/glew -isystem /usr/local/Cellar/libogg/1.3.5/include -isystem /usr/local/Cellar/opusfile/0.12/include/opus -isystem /usr/local/Cellar/opus/1.3.1/include/opus -isystem /Library/Frameworks/Mono.framework/Headers -isystem /Users/runner/work/ddnet/ddnet/ddnet-libs/sdl/include/mac -isystem /Users/runner/work/ddnet/ddnet/src/engine/external/wavpack -isystem /usr/local/Cellar/ffmpeg/5.1/include -Werror -g -isysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk -mmacosx-version-min=10.13 -F/Users/runner/work/ddnet/ddnet/ddnet-libs/sdl/mac/lib64  -fdiagnostics-color=always -fcolor-diagnostics -fstack-protector-all -fno-exceptions -stdlib=libc++ -std=gnu++17 -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wshadow-all -Wthread-safety -Wthread-safety-negative -Wsuggest-override -std=c++17 -MD -MT CMakeFiles/DDNet.dir/src/engine/client/video.cpp.o -MF CMakeFiles/DDNet.dir/src/engine/client/video.cpp.o.d -o CMakeFiles/DDNet.dir/src/engine/client/video.cpp.o -c /Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:622:10: error: 'channel_layout' is deprecated [-Werror,-Wdeprecated-declarations]
        pFrame->channel_layout = ChannelLayout;
                ^
/usr/local/Cellar/ffmpeg/5.1/include/libavutil/frame.h:509:5: note: 'channel_layout' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/Cellar/ffmpeg/5.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:742:81: error: 'channel_layout' is deprecated [-Werror,-Wdeprecated-declarations]
                m_AudioStream.m_vpFrames[i] = AllocAudioFrame(pContext->sample_fmt, pContext->channel_layout, pContext->sample_rate, NbSamples);
                                                                                              ^
/usr/local/Cellar/ffmpeg/5.1/include/libavcodec/avcodec.h:1059:5: note: 'channel_layout' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/Cellar/ffmpeg/5.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:782:79: error: 'channels' is deprecated [-Werror,-Wdeprecated-declarations]
                av_opt_set_int(m_AudioStream.m_vpSwrCtxs[i], "out_channel_count", pContext->channels, 0);
                                                                                            ^
/usr/local/Cellar/ffmpeg/5.1/include/libavcodec/avcodec.h:1005:5: note: 'channels' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/Cellar/ffmpeg/5.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:850:13: error: 'channels' is deprecated [-Werror,-Wdeprecated-declarations]
                pContext->channels = 2;
                          ^
/usr/local/Cellar/ffmpeg/5.1/include/libavcodec/avcodec.h:1005:5: note: 'channels' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/Cellar/ffmpeg/5.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:851:13: error: 'channel_layout' is deprecated [-Werror,-Wdeprecated-declarations]
                pContext->channel_layout = AV_CH_LAYOUT_STEREO;
                          ^
/usr/local/Cellar/ffmpeg/5.1/include/libavcodec/avcodec.h:1059:5: note: 'channel_layout' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/usr/local/Cellar/ffmpeg/5.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
5 errors generated.
```

https://github.com/ddnet/ddnet/runs/7591197768?check_suite_focus=true#step:7:344

--- a/src/engine/client/video.cpp
+++ b/src/engine/client/video.cpp
@@ -619,7 +619,11 @@ AVFrame *CVideo::AllocAudioFrame(enum AVSampleFormat SampleFmt, uint64_t Channel
 	}
 
 	pFrame->format = SampleFmt;
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(59, 24, 100)
+	dbg_assert(av_channel_layout_from_mask(&pFrame->ch_layout, ChannelLayout) == 0, "Failed to set channel layout");
+#else
 	pFrame->channel_layout = ChannelLayout;
+#endif
 	pFrame->sample_rate = SampleRate;
 	pFrame->nb_samples = NbSamples;
 
@@ -739,7 +743,11 @@ bool CVideo::OpenAudio()
 	for(size_t i = 0; i < m_AudioThreads; ++i)
 	{
 		m_AudioStream.m_vpFrames.emplace_back(nullptr);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(59, 24, 100)
+		m_AudioStream.m_vpFrames[i] = AllocAudioFrame(pContext->sample_fmt, pContext->ch_layout.u.mask, pContext->sample_rate, NbSamples);
+#else
 		m_AudioStream.m_vpFrames[i] = AllocAudioFrame(pContext->sample_fmt, pContext->channel_layout, pContext->sample_rate, NbSamples);
+#endif
 		if(!m_AudioStream.m_vpFrames[i])
 		{
 			dbg_msg("video_recorder", "Could not allocate audio frame");
@@ -779,7 +787,11 @@ bool CVideo::OpenAudio()
 		av_opt_set_int(m_AudioStream.m_vpSwrCtxs[i], "in_channel_count", 2, 0);
 		av_opt_set_int(m_AudioStream.m_vpSwrCtxs[i], "in_sample_rate", g_Config.m_SndRate, 0);
 		av_opt_set_sample_fmt(m_AudioStream.m_vpSwrCtxs[i], "in_sample_fmt", AV_SAMPLE_FMT_S16, 0);
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(59, 24, 100)
+		av_opt_set_int(m_AudioStream.m_vpSwrCtxs[i], "out_channel_count", pContext->ch_layout.nb_channels, 0);
+#else
 		av_opt_set_int(m_AudioStream.m_vpSwrCtxs[i], "out_channel_count", pContext->channels, 0);
+#endif
 		av_opt_set_int(m_AudioStream.m_vpSwrCtxs[i], "out_sample_rate", pContext->sample_rate, 0);
 		av_opt_set_sample_fmt(m_AudioStream.m_vpSwrCtxs[i], "out_sample_fmt", pContext->sample_fmt, 0);
 
@@ -847,8 +859,12 @@ bool CVideo::AddStream(OutputStream *pStream, AVFormatContext *pOC, const AVCode
 				}
 			}
 		}
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(59, 24, 100)
+		dbg_assert(av_channel_layout_from_mask(&pContext->ch_layout, AV_CH_LAYOUT_STEREO) == 0, "Failed to set channel layout");
+#else
 		pContext->channels = 2;
 		pContext->channel_layout = AV_CH_LAYOUT_STEREO;
+#endif
 
 		pStream->pSt->time_base.num = 1;
 		pStream->pSt->time_base.den = pContext->sample_rate;

